### ISSUE TITLE
Provide lane advice on junction=circular

### ISSUE THREAD

**Issue by tommycrock** [osmandapp/OsmAnd#10907]
# 🚀 feature request

### Description

When navigating on circulars, no lane advice is given when approaching or on the circular (same may be true for roundabouts?), only the relative exit number is displayed. This can be misleading and difficult to follow. Circulars are often some of the hardest junctions to navigate so these should be a priority for lane advice. Discussed on the [Google Groups forum](https://groups.google.com/g/osmand/c/E7cKJxRSUe4) and advised to open a feature request via support@osmand.net

### Describe the solution you'd like

Lane advice should be provided on circulars where lanes are marked correctly in OSM. I appreciate this can be complicated to interpret, and I don't know if [placement tags](https://wiki.openstreetmap.org/wiki/Key:placement) and [connectivity relations](https://wiki.openstreetmap.org/wiki/Relation:connectivity) are currently evaluated to help with lane guidance? Using one or both of these approaches would be great. Ideally, current and next lanes should be displayed where necessary, as rapid changes can be needed, so nice to be prepared.

Example of the problem:

![Screenshot_20210216-121402_OsmAnd+](https://user-images.githubusercontent.com/5749884/108061990-0dbbb780-7051-11eb-98da-ec2e780908da.jpg)

Aerial imagery for example:
![image](https://user-images.githubusercontent.com/5749884/108063088-9dae3100-7052-11eb-8b07-8aed27524814.png)
from https://www.openstreetmap.org/edit#map=18/53.87190/-1.70874

suggested lane assistance mock-up
![image](https://user-images.githubusercontent.com/5749884/108062727-211b5280-7052-11eb-9294-c0cbcb55e18f.png)

### Describe alternatives you've considered

#3638 suggests a new tag (circleexit:lanes). This may be redundant with the above and would require a definition of which ways should be counted. It also wouldn't tell you when to change lanes on the circular where necessary (unless I've misunderstood).

**Comment by tommycrock** [comment]
In case it wasn't clear, there are probably three separable improvements here.
1. Provide lane advice on circulars / roundabouts when it can be reasonably determined. I would have thought this was relatively straightforward for the exits for most as they should be separated, so not much different from most junctions?
2. Improve osmand's interpretation of lanes to enable accurate advice in more complicated situations.*
3. Provide now and next lanes. This is probably just when next (different) set of lanes is within some expected time of arrival based on speed and distance, say 3 seconds away? In my example there's at least four different sets of lanes on the route around the circular, probably five or six.
* This is based on my assumption that osmand doesn't account for placement or connectivity, but perhaps it does, in which case we should add more of those to OSM!

**Comment by NickKatchur** [comment]
Hoping to bring this back to active discussion. There is still currently no roundabout turn lane advice given. I believe there is sufficient lane and connectivity information in OSM for this to happen. Such as that I've added to this intersection: 38.34455,-85.70576

**Comment by aceman444** [comment]
Yes this is very important to solve. Support connectivity relations (issue #18853) and turn lanes inside roundabouts.
Nowadays even crazier turbo roundabouts are being created in the wild (see https://wiki.openstreetmap.org/wiki/Tag:roundabout%3Dturbo ) and you may need to know with which lane to even enter the roundabout so that it then allows you to exit it at the proper exit, as you cannot always change lanes inside the roundabout freely.
You can see a case e.g. at https://www.openstreetmap.org/#map=19/48.213464/17.220933&layers=N .
